### PR TITLE
New constant string format specifier for sprintf

### PR DESCRIPTION
**Pull request by t-wissmann**
This introduces a new format specifier '%c' in the sprintf format
strings. Of course it never makes sense to manually write something like

    hc sprintf S '%c.name' 'tags.0'

but it makes sense in combination with the other commands that replace
certain identifiers in commands, especially in combination with the foreach
command.

This change required an update in the completion code for the 'sprintf'
command, and while at it, I fixed a minor bug in the completion. The bug
was that completing the command

   hc sprintf S '%s'

erroneously yielded S.

### CODE REVIEW

**Comment by blarz** [comment]
Fixed some typos in the PR description.

File: src/rootcommands.cpp [hunk 1]
```diff
@@ -237,6 +239,14 @@ int RootCommands::sprintf_cmd(Input input, Output output)
     for (const auto& blob : format) {
         if (blob.literal_) {
             replacedString += blob.data_;
+        } else if (blob.data_ == "c"){
```
Comment (blarz) [inline src/rootcommands.cpp:242]: Style: Add a space between ) and {

File: src/rootcommands.cpp [hunk 2]
```diff
@@ -262,22 +272,30 @@ void RootCommands::sprintf_complete(Completion& complete)
...
+            fs = parseFormatString(complete[1]);
         }  catch (const std::invalid_argument&) {
             complete.invalidArguments();
+            return;
         }
-        if (complete < 2 + placeholderCount) {
-            completeAttributePath(complete);
-        } else {
-            complete.completeCommands(2 + placeholderCount);
+        int	indexOfNextArgument = 2;
```
Comment (blarz) [inline src/rootcommands.cpp:282]: Double whitespace

**Review by blarz — CHANGES_REQUESTED** [review]

**Comment by t-wissmann** [comment]
I've fixed both whitespace issues (the 'double' whitespace was a tab; I don't know how qtcreator was able to sneak that in)

**Review by blarz — DISMISSED** [review]
Nice! Looking forward to see what the users will do with this :D

**Comment by t-wissmann** [comment]
I've pushed a new commit resolving the merge conflicts in `doc/herbstluftwm.txt` that arised from the merge of the foreach-PR.

As soon as this is merged, I can answer a few open issues/feature requests :-) (closing all clients is one of them)

**Review by blarz — APPROVED** [review]

**Pull request merged**